From c725387502629782970a262a0d78bfb50c7dd0b3 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Tue, 21 Feb 2017 22:26:27 +0100
Subject: [PATCH] iplink: bridge_slave: add support for IFLA_BRPORT_FAST_LEAVE

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1374360
Upstream Status: iproute2.git commit 478a8e5920684

commit 478a8e59206841b2a2d32a29fdd0cbec1965cd5e
Author: Nikolay Aleksandrov <nikolay@cumulusnetworks.com>
Date:   Tue Feb 16 16:08:55 2016 +0100

    iplink: bridge_slave: add support for IFLA_BRPORT_FAST_LEAVE

    Add support to be able to view and change IFLA_BRPORT_FAST_LEAVE
    port attribute.

    Signed-off-by: Nikolay Aleksandrov <nikolay@cumulusnetworks.com>
---
 ip/iplink_bridge_slave.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/ip/iplink_bridge_slave.c b/ip/iplink_bridge_slave.c
index 9ed44b1..66c3a00 100644
--- a/ip/iplink_bridge_slave.c
+++ b/ip/iplink_bridge_slave.c
@@ -30,6 +30,7 @@ static void print_explain(FILE *f)
 		"                        [ learning {on | off} ]\n"
 		"                        [ flood {on | off} ]\n"
 		"                        [ mcast_router MULTICAST_ROUTER ]\n"
+		"                        [ mcast_fast_leave {on | off} ]\n"
 	);
 }
 
@@ -103,6 +104,10 @@ static void bridge_slave_print_opt(struct link_util *lu, FILE *f,
 	if (tb[IFLA_BRPORT_MULTICAST_ROUTER])
 		fprintf(f, "mcast_router %u ",
 			rta_getattr_u8(tb[IFLA_BRPORT_MULTICAST_ROUTER]));
+
+	if (tb[IFLA_BRPORT_FAST_LEAVE])
+		print_onoff(f, "mcast_fast_leave",
+			    rta_getattr_u8(tb[IFLA_BRPORT_FAST_LEAVE]));
 }
 
 static void bridge_slave_parse_on_off(char *arg_name, char *arg_val,
@@ -175,6 +180,10 @@ static int bridge_slave_parse_opt(struct link_util *lu, int argc, char **argv,
 				invarg("invalid mcast_router", *argv);
 			addattr8(n, 1024, IFLA_BRPORT_MULTICAST_ROUTER,
 				 mcast_router);
+		} else if (matches(*argv, "mcast_fast_leave") == 0) {
+			NEXT_ARG();
+			bridge_slave_parse_on_off("mcast_fast_leave", *argv, n,
+						  IFLA_BRPORT_FAST_LEAVE);
 		} else if (matches(*argv, "help") == 0) {
 			explain();
 			return -1;
-- 
1.8.3.1

